lib/widget/registration/create_organisation_step/create_organisation_widget.dart

```diff
@@ -0,0 +1,268 @@
+import 'package:logotime/widget/common/buttons/regular_button.dart';
+import 'package:logotime/widget/common/forms/labeled_switch.dart';
+import 'package:logotime/widget/common/forms/radio_buttons.dart';
+import 'package:logotime/widget/common/forms/styled_text_field.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:logotime/widget/registration/create_organisation_step/create_organisation_view_model.dart';
+
+class CreateOrganisationWidget extends StatefulWidget {
+  final CreateOrganisationViewModel viewModel;
+
+  const CreateOrganisationWidget({super.key, required this.viewModel});
+
+  @override
+  State<CreateOrganisationWidget> createState() =>
+      _CreateOrganisationWidgetState();
+}
+
+class _CreateOrganisationWidgetState extends State<CreateOrganisationWidget> {
+  late TextEditingController _organisationNameTextController;
+  late TextEditingController _maxApplicationTextController;
+  OrganisationSize? _organisationSize;
+
+  @override
+  void initState() {
+    super.initState();
+    _organisationNameTextController = TextEditingController();
+    _maxApplicationTextController = TextEditingController(
+      text: '${widget.viewModel.organisationRules.maxApplications}',
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final localizations = AppLocalizations.of(context)!;
+    final rules = widget.viewModel.organisationRules;
+    if (rules.unassignedShiftRule == UnassignedShiftRule.prohibited) {
+      _maxApplicationTextController.text = '0';
+    }
+    return Scaffold(
+      appBar: AppBar(
+        title: Text(localizations.createOrganisation.capitalize()),
+      ),
+      body: SafeArea(
+        child: Padding(
+          padding: const EdgeInsets.symmetric(
+            horizontal: 30.0,
+          ),
+          child: SingleChildScrollView(
+            child: Column(
+              crossAxisAlignment: CrossAxisAlignment.stretch,
+              children: [
+                Padding(
+                  padding: const EdgeInsets.only(top: 30.0),
+                  child: StyledTextField(
+                    fieldTitle: localizations.organisationName.capitalize(),
+                    controller: _organisationNameTextController,
+                    error: widget.viewModel.organisationNameError,
+                  ),
+                ),
+                Text(
+                  localizations.organisationSize.capitalize(),
+                  textAlign: TextAlign.center,
+                  style: const TextStyle(
+                    fontSize: 25.0,
+                  ),
+                ),
+                const SizedBox(
+                  height: 20.0,
+                ),
+                Row(
+                  mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                  children: _organisationSizeButtons(),
+                ),
+                Visibility(
+                  visible: widget.viewModel.organisationSizeError != null,
```

Comment:
same comment as for previous project: I wouldn't use Visibility at all. `if` can do this job